### PR TITLE
[Deno] Database migration

### PR DESCRIPTION
**Pull request by vyktoremario**
## Type of change

- [ ] Documentation change
- [ ] Bug fix
- [x]  Feature

## Summary of change

Closes: #605 

## Checklist

- [x] These changes follow the [contributing guidelines](https://github.com/thisdot/starter.dev/blob/main/CONTRIBUTING.md)
- [x] This starter kit has been approved by the maintainers
- [x] Changes for this new starter kit are being pushed to an integration branch instead of main
- [x] All dependencies are version locked
- [ ] This fix resolves #
- [ ] I have verified the fix works and introduces no further errors

### CODE REVIEW

File: starters/deno-oak-denodb/src/db/resolvers/technology.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import { Technologies } from '../model/technology.ts';
+
+export interface ITechnology {
+	displayName: string;
+	description: string;
+	url: string;
+}
+
+export const technologyResolvers = {
+	Query: {
```
Comment (dariodjuric) [inline starters/deno-oak-denodb/src/db/resolvers/technology.ts:10]: Do you think we could put query handlers to a separate module, as well as mutations to its own file?

File: starters/deno-oak-denodb/src/main.ts [hunk 2]
```diff
@@ -1,24 +1,38 @@
...
+	}),
 );
+
+app.use(GraphQLService.routes(), GraphQLService.allowedMethods());
 app.use(router.routes());
 app.use(router.allowedMethods());
 
+await db.sync({ drop: true });
+
+console.log(`Database connected!`);
```
Comment (dariodjuric) [inline starters/deno-oak-denodb/src/main.ts:35]: What do you think about logging the hostname of the database as well?

File: starters/deno-oak-denodb/src/main.ts [hunk 3]
```diff
@@ -1,24 +1,38 @@
...
+		origin: CORS_ALLOWED_ORIGINS ? CORS_ALLOWED_ORIGINS.split(',') : '',
+		optionsSuccessStatus: 200, // Some legacy browsers (IE11, various SmartTVs) choke on 204
+	}),
 );
+
+app.use(GraphQLService.routes(), GraphQLService.allowedMethods());
 app.use(router.routes());
 app.use(router.allowedMethods());
 
+await db.sync({ drop: true });
```
Comment (dariodjuric) [inline starters/deno-oak-denodb/src/main.ts:33]: This is a dangerous operation to be in the app startup module, especially with `drop: true`. I would suggest to place it in its own module.

File: starters/deno-oak-denodb/src/db/resolvers/technology.ts (hunk 1 above)
Comment (vyktoremario) [reply starters/deno-oak-denodb/src/db/resolvers/technology.ts:10]: I have moved it, but I am not sure if that's what you mean

File: starters/deno-oak-denodb/src/main.ts (hunk 3 above)
Comment (dariodjuric) [reply starters/deno-oak-denodb/src/main.ts:34]: For example, you can place it in `seed.ts` and then the seed command would be `deno run seed.ts ...`. It's a manual command that needs to be executed during project setup.

File: starters/deno-oak-denodb/src/db/model/technology.ts [hunk 4]
```diff
@@ -0,0 +1,16 @@
+import { DataTypes, Model } from '../../../deps.ts';
+
+export class Technologies extends Model {
```
Comment (dariodjuric) [inline starters/deno-oak-denodb/src/db/model/technology.ts:3]: Shouldn't we be using a `User` model here?

File: starters/deno-oak-denodb/src/main.ts (hunk 3 above)
Comment (dariodjuric) [reply starters/deno-oak-denodb/src/main.ts:34]: Oh, I see you already have `run_seeder.ts`. Yeah, that's good.

So you can remove this line then.

File: starters/deno-oak-denodb/src/db/model/technology.ts (hunk 4 above)
Comment (dariodjuric) [reply starters/deno-oak-denodb/src/db/model/technology.ts:3]: Or if we have Technologies in other starter kit, we can use that, sure.

File: starters/deno-oak-denodb/src/db/run_seeder.ts [hunk 5]
```diff
@@ -0,0 +1,30 @@
...
+import { technologySeedData } from './seeding/technology-seeder.ts';
+
+import {
+	DATABASE_HOST,
+	DATABASE_NAME,
+	DATABASE_PASSWORD,
+	DATABASE_USERNAME,
+} from '../config/environment.ts';
+
+async function runSeeder() {
```
Comment (dariodjuric) [inline starters/deno-oak-denodb/src/db/run_seeder.ts:12]: Could we add an instruction in the readme on how to run the seeder?

File: starters/deno-oak-denodb/src/db/resolvers/mutationHandler.ts [hunk 6]
```diff
@@ -0,0 +1,38 @@
+import { Technologies } from '../model/technology.ts';
+export interface ITechnology {
```
Comment (dariodjuric) [inline starters/deno-oak-denodb/src/db/resolvers/mutationHandler.ts:2]: I'm not clear on why this interface is needed. Should it correspond to the GraphQL type `Technology`?

**Review by dariodjuric — CHANGES_REQUESTED** [review]
I left some questions/comments.

File: starters/deno-oak-denodb/src/db/model/technology.ts (hunk 4 above)
Comment (vyktoremario) [reply starters/deno-oak-denodb/src/db/model/technology.ts:3]: Based on the Document Dustin shared, this was the model he said the backend kits should have

File: starters/deno-oak-denodb/src/db/resolvers/mutationHandler.ts (hunk 6 above)
Comment (vyktoremario) [reply starters/deno-oak-denodb/src/db/resolvers/mutationHandler.ts:2]: This corresponds to the `TechnologyInput` in the type definitions.  
I can use `any` for now

File: starters/deno-oak-denodb/src/db/model/technology.ts (hunk 4 above)
Comment (dariodjuric) [reply starters/deno-oak-denodb/src/db/model/technology.ts:3]: Ah, okay. I missed that. 🙊

File: starters/deno-oak-denodb/src/db/resolvers/mutationHandler.ts (hunk 6 above)
Comment (dariodjuric) [reply starters/deno-oak-denodb/src/db/resolvers/mutationHandler.ts:2]: I see. In that case, I would suggest to name it that way: `TechnologyInput`. You could also place it in its own folder under the GraphQL API folder, e.g. `types`.

Which reminds me, ne other thing that we might consider is splitting the GraphQL-related code into its own folder. Right now it's located under the `db` folder, but the GraphQL code is really database-independent. What do you think about having a `graphql` folder, which would host API types like `TechnologyInput`, the schema, resolvers, etc.

Comment (dariodjuric) [reply starters/deno-oak-denodb/src/db/resolvers/mutationHandler.ts]: Can we generate those somehow?

If so, we should have this documented in the readme on how to do it. The initial types should be generated by us.

Comment (vyktoremario) [reply starters/deno-oak-denodb/src/db/resolvers/mutationHandler.ts]: For now, let's leave things as they are. 
I will be trying that in a different PR

**Review by dariodjuric — APPROVED** [review]

**Pull request merged**